src/test/java/com/nedap/openehr/lsp/RunCommandTest.java

```diff
@@ -0,0 +1,115 @@
+package com.nedap.openehr.lsp;
+
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+public class RunCommandTest extends LanguageServerTestBase {
+
+
+    /**
+     * A parameterized test that tests generating examples and opts
+     * @param archetypeToTest the filename of the archetype to test
+     * @param otherArchetypeNames file names of other archetypes to load before the archetypeToTest
+     * @param commandIdentifier the identifier of the command to execute, for example source.example
+     * @param commandKind the kind of the command: source.example.json for example
+     * @param createdFileName the file name that should be created
+     * @param startsWith the first characters of the created file
+     * @throws Exception in case something goes wrong and the test fails :)
+     */
+    @ParameterizedTest
+    @MethodSource("exampleArchetypeProvider")
+    public void generateFile(String archetypeToTest,
+                                List<String> otherArchetypeNames,
+                                String commandIdentifier,
+                                String commandKind,
+                                String createdFileName,
+                                String startsWith) throws Exception {
+        for(String otherArchetype:otherArchetypeNames) {
+            openResource(otherArchetype);
+        }
+        openResource(archetypeToTest);
+
+        //check that the archetypes are correct
+        assertTrue(testClient.getDiagnostics().get(archetypeToTest).getDiagnostics().isEmpty());
+        for(String otherArchetype:otherArchetypeNames) {
+            assertTrue(testClient.getDiagnostics().get(otherArchetype).getDiagnostics().isEmpty());
+        }
```

Comment:
That is not a good idea. openResource can fail if you assert directly after opening resources, due to missing dependencies. Then if you add the missing dependency by opening the other resource, the server will automatically fix these by pushing new diagnostics, which are then correctly handled in the testclient by removing the old one. Your change would make the test depend on the ordering of the loaded archetypes, and that is not what this test is testing.

I will add the diagnostics to the error message, which is a minor improvement since really this is only to check if the author of the test archetypes did not make a mistake, not if validation or diagnostics work properly.

Reply:
Makes sense. Thanks for the good explanation. 👍🏻